addons/point_of_sale/static/src/app/services/pos_store.js

```diff
@@ -506,9 +506,15 @@ export class PosStore extends Reactive {
         };
     }
     getDefaultSearchDetails() {
+        let field = "RECEIPT_NUMBER";
+        let term = "";
+        if (this.get_order()?.get_partner()) {
+            field = "PARTNER";
+            term = this.get_order().get_partner().name;
+        }
         return {
-            fieldName: "RECEIPT_NUMBER",
-            searchTerm: "",
+            fieldName: field,
+            searchTerm: term,
```

Comment:
I think this code is useless since we have this:

```js
    goToOrders(partner) {
        this.props.close();
        const partnerHasActiveOrders = this.pos
            .get_open_orders()
            .some((order) => order.partner?.id === partner.id);
        const stateOverride = {
            search: {
                fieldName: "PARTNER",
                searchTerm: partner.name,
            },
            filter: partnerHasActiveOrders ? "" : "SYNCED",
        };
        this.pos.showScreen("TicketScreen", { stateOverride });
    }
```

Reply:
How is it useless? Can you please specify what you mean?

Reply:
Mybad, I was thinking the auto selection was only needed when accessing ticket_screen from customer popup, so it LGTM :)